tests/components/nuheat/test_climate.py

```diff
@@ -135,3 +140,23 @@ async def test_climate_thermostat_schedule_temporary_hold(hass):
     # Only test for a subset of attributes in case
     # HA changes the implementation and a new one appears
     assert all(item in state.attributes.items() for item in expected_attributes.items())
+
+    await hass.services.async_call(
+        "climate",
+        "set_temperature",
+        service_data={ATTR_ENTITY_ID: "climate.temp_bathroom", "temperature": 90},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    # opportunistic set
+    state = hass.states.get("climate.temp_bathroom")
+    assert state.attributes["preset_mode"] == "Temporary Hold"
+    assert state.attributes["temperature"] == 50.0
+
+    # and the api poll returns it to the mock
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=3))
+    await hass.async_block_till_done()
+    state = hass.states.get("climate.temp_bathroom")
+    assert state.attributes["preset_mode"] == "Run Schedule"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)